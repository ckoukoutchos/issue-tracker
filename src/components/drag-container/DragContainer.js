import React, { useState } from "react";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
import styles from './DragContainer.module.css';
import Avatar from '../avatar/Avatar';
import Card from '../card/Card';

const DragContainer = ({ items, updateOrder }) => {
  const reorder = (list, startIndex, endIndex) => {
    const result = Array.from(list);
    const [removed] = result.splice(startIndex, 1);
    result.splice(endIndex, 0, removed);
    return result;
  };

  const onDragEnd = (result) => {
    if (!result.destination) {
      return;
    }

    const reorderedItems = reorder(
      items,
      result.source.index,
      result.destination.index
    );
    updateOrder(reorderedItems);
  }

  return (
    <DragDropContext onDragEnd={onDragEnd}>
      <Droppable droppableId="droppable">
        {(provided, snapshot) => (
          <div
            {...provided.droppableProps}
            ref={provided.innerRef}
          >
            {items.map((item, index) => (
              <Draggable key={item.id} draggableId={String(item.id)} index={index}>
                {(provided, snapshot) => (
                  <div
                    ref={provided.innerRef}
                    {...provided.draggableProps}
                    {...provided.dragHandleProps}
                  >
                    <Card>
                      <div className={styles.container}>
                        <Avatar url={item.avatar} />
                        <p className={styles.title}>{item.title}</p>
                      </div>
                      <p className={styles.text}>Created: {item.created}</p>
                      <p className={styles.text}>Last Updated: {item.updated}</p>
                    </Card>
                  </div>
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  );
}

export default DragContainer;
