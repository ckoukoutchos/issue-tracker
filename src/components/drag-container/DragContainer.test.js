import React from 'react';
import DragContainer from './DragContainer';
import renderer from 'react-test-renderer';

it('renders the DragContainer component correctly', () => {
  const tree = renderer
    .create(<DragContainer items={[]} updateOrder={() => { }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});