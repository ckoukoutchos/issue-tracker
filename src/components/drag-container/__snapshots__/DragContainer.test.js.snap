// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DragContainer component correctly 1`] = `
<div
  data-rbd-droppable-context-id="0"
  data-rbd-droppable-id="droppable"
/>
`;
